src/shared_modules/rsync/src/rsyncImplementation.cpp

```diff
@@ -150,21 +152,34 @@ void RSyncImplementation::registerSyncId(const RSYNC_HANDLE handle,
 
     const auto registerCallback
     {
-        [spDBSyncWrapper, syncConfiguration, callbackWrapper] (const SyncInputData & syncData)
+        [spDBSyncWrapper, syncConfiguration, callbackWrapper, handle] (const SyncInputData & syncData)
         {
             try
             {
-                if (0 == syncData.command.compare("checksum_fail"))
+                if (syncData.id < SynchronizationController::instance().get(handle))
                 {
-                    sendChecksumFail(spDBSyncWrapper, syncConfiguration, callbackWrapper, syncData);
+                    SynchronizationController::instance().set(handle, syncData.id);
                 }
-                else if (0 == syncData.command.compare("no_data"))
+
+                if (syncData.id > SynchronizationController::instance().get(handle))
```

Comment:
Question: would it be possible that the values of the map change  between each  `syncData.id < SynchronizationController::instance().get(handle)`  call? Should `registerCallback` be atomic?

Reply:
Good catch! yes, this is is probably a better option.